log-manager/src/main/java/io/airlift/log/LogFileName.java

```diff
@@ -0,0 +1,255 @@
+ * limitations under the License.
+ */
+package io.airlift.log;
+
+import com.google.common.base.Splitter;
+import com.google.common.collect.ComparisonChain;
+
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.time.DateTimeException;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.OptionalInt;
+
+import static com.google.common.base.CharMatcher.anyOf;
+import static java.lang.Integer.parseInt;
+import static java.util.Objects.requireNonNull;
+import static java.util.UUID.randomUUID;
+
+class LogFileName
+        implements Comparable<LogFileName>
+{
+    private static final DateTimeFormatter DATE_TIME_FORMATTER = DateTimeFormatter.ofPattern("-yyyyMMdd.HHmmss");
+    private static final int MAX_GENERATED_INDEX = 1000;
+
+    private final String fileName;
+    private final LocalDateTime dateTime;
+    private final OptionalInt index;
+    private final OptionalInt legacyIndex;
+    private final Optional<String> slug;
+    private final boolean compressed;
+
+    /**
+     * Attempts to parse a log file name that is part of the history files for the master log file.
+     *
+     * @param masterLogFileName the main file of the logger
+     * @param historyFileName the history file name to parse
+     * @return a present value if the file name is part of the history set
+     */
+    public static Optional<LogFileName> parseHistoryLogFileName(String masterLogFileName, String historyFileName)
+    {
+        requireNonNull(masterLogFileName, "masterLogFileName is null");
+        requireNonNull(historyFileName, "historyFileName is null");
+        if (!historyFileName.startsWith(masterLogFileName + "-")) {
+            return Optional.empty();
+        }
+
+        String remainder = historyFileName.substring(masterLogFileName.length() + 1);
+
+        boolean compressed = remainder.endsWith(".gz");
+        if (compressed) {
+            remainder = remainder.substring(0, remainder.length() - ".gz".length());
+        }
+
+        if (remainder.isEmpty()) {
+            // This is not considered a valid log file, because the log should have already been renamed
+            return Optional.empty();
+        }
+
+        if (remainder.endsWith(".log")) {
+            // %d{yyyy-MM-dd}.%i.log
+            remainder = remainder.substring(0, remainder.length() - ".log".length());
+
+            List<String> parts = Splitter.on(anyOf(".-")).limit(4).splitToList(remainder);
+            if (parts.size() != 4) {
+                return Optional.empty();
+            }
+
+            LocalDateTime dateTime;
+            try {
+                dateTime = LocalDateTime.of(parseInt(parts.get(0)), parseInt(parts.get(1)), parseInt(parts.get(2)), 0, 0);
+            }
+            catch (NumberFormatException | DateTimeException e) {
+                return Optional.empty();
```

Comment:
a single regex 
```
(?<date>\d\d\d\d-\d\d-\d\d).\d.log(?<compressed>.gz)?
```

would yield to shorter & more readable code.

Is this piece considered performance critical?